### DISCUSSION TITLE
scheduleRecurrently + jdbctemplate = java.lang.IllegalStateException: No DataSource set

### DISCUSSION THREAD

**Discussion by rjhho** [jobrunr/jobrunr#1112] [Q&A]
Maven version: apache-maven-3.6.3
jdk: 1.8
springboot: 2.7.18

Hi Jobrunr community,
I'm encountering the following issue during an in-memory integration test, :

I'm trying to update a column called "message" within a table called "berichten" and change the "message value from "TEST" to "TEST2".
By creating the following method: 
```
        String sql = "UPDATE Berichten SET message='TEST2'";

        BackgroundJob.scheduleRecurrently("id_taak_1", Cron.every15seconds(), () -> {
            jdbcTemplate.update(sql);
        });
```
Unfortunately i get the following error:

`WARN org.jobrunr.server.BackgroundJobPerformer - Job(id=0191dc57-2a2e-7267-ab07-649fc7bdfedd, jobName='org.springframework.jdbc.core.JdbcTemplate.update(UPDATE Berichten SET id=420)') processing failed: An exception occurred during the performance of the job
java.lang.IllegalStateException: No DataSource set`

The jdctemplate itself seems to work just fine if i call jdbcTemplate.update(sqlquery) without using schedulerecurrently.
Can you guys please help me with this issue?

Here is the simple project to reproduce the issue
https://github.com/rjhho/RecurrentlyTestProject 

Thanks in advance! 🙏 

Raymond

**Comment by rdehuyss** [comment]
You need to wrap your call to the jdbcTemplate in a Spring service where you then do the `jdbcTemplate.update(sql)` or need to use the `BackgroundJobRequest`.

**Reply by rjhho** [reply]
Thank you for your time @rdehuyss
I've moved the jdbcTemplate into a Spring service class, the changes are pushed into https://github.com/rjhho/RecurrentlyTestProject 

```
@Service
public class BerichtService {

    @Autowired
    private JdbcTemplate jdbcTemplate;

    public int updateBericht() {
        System.out.println("jdbctemplate is: " + jdbcTemplate);
        System.out.println("code has been reached aka updateBericht");
        String sql = "UPDATE Berichten SET message='TEST2'";
        return jdbcTemplate.update(sql);
    }
}
```

It seems to work fine when i just call the "updateBericht" method, unfortunately when i call it through 
`BackgroundJob.scheduleRecurrently("*/5 * * * * *", () -> berichtService.updateBericht());
` the jdbctemplate seems to be null for some reason

Do you know what's going on here?

**Reply by rdehuyss** [reply]
Your test setup is initializing JobRunr so you have 2 BackgroundJobServers running - one created by the `jobrunr-spring-boot-2-starter` and one by you. The one you have setup does not know about the JDBC Template as there is no `JobActivator`.

**Reply by rjhho** [reply]
It works! Thanks for your help and guidance!
